gulpfile.ts

```diff
@@ -0,0 +1,243 @@
+import gulp from 'gulp';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const clean = require('gulp-clean');
+import merge from 'merge-stream';
+import ts from 'gulp-typescript';
+import sourcemaps from 'gulp-sourcemaps';
+import terser from 'gulp-terser';
+import uglify from 'gulp-uglify';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const rollup = require('gulp-rollup-2').rollup;
+import rollupTypescript from 'rollup-plugin-typescript2';
+
+import minifyPrivatesTransformer from 'ts-transformer-minify-privates';
+import tsMacroTransformer from 'ts-macros';
+import pathsTransformer from 'ts-transform-paths';
+import { CustomTransformerFactory } from 'typescript';
+
+gulp.task('cjs', () => {
+  const tsCjsProject = ts.createProject('tsconfig.json', {
```

Comment:
Can some generic functions be extracted here to make pipe simple?

Reply:
Are you saying that it is better to make the overlapping parts of each build into a function?

Reply:
Or split into multiple files?

Reply:
> Are you saying that it is better to make the overlapping parts of each build into a function?

It would be nice if could. 👍

Reply:
I modified it to look much better.